### ISSUE TITLE
Create a custom single user image

### ISSUE THREAD

**Issue by betatim** [wildtreetech/ohjh#4]
We want to have some custom libraries and tools installed in the single user image.

Things we need for sure:

* server extension (#3)
* [OH API library](https://pypi.python.org/pypi/open-humans-api)
* R kernel (and R studio via [nbrsessionproxy](https://github.com/jupyterhub/nbrsessionproxy)?)
* httpimport
* myvariant

More ideas?

**Comment by betatim** [comment]
We can test drive this image via mybinder.org and this repository I think.

**Comment by betatim** [comment]
Current image is: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/tree/master/images/singleuser-sample

**Comment by madprime** [comment]
anaconda? Well, that's not really a library, it's a bunch. I think @gedankenstuecke is better at data analysis in general, and especially in R, and is more familiar with libraries that would be desired.

**Comment by gedankenstuecke** [comment]
I've never tried the R kernel, but can users install packages in it using `install.packages()` or `devtools` in their own notebooks? Because in that case we wouldn't need to provide any packages for R I think. Installing packages with these two methods is pretty straight forward.

**Comment by madprime** [comment]
I think "R kernel" just means the ability to do a Jupyter notebook in R. https://github.com/IRkernel/IRkernel And yes, we should start by having that! RStudio sounds good to me but I don't speak R.

Would devtools itself be a package? (or maybe it's a default thing?)

**Comment by gedankenstuecke** [comment]
Yeah, but if users get full control of their R session they can just install stuff on their own. `devtools` is its own package (which might be useful to be preinstalled) but it can also be just installed with `install.packages()` if users can do this.

**Comment by madprime** [comment]
I may misunderstand, but I suspect pre-installation of libraries/packages/tools on the user image is in part desirable as a time-saver – assuming installation of these things by each user (even if theoretically doable) takes any amount of time to perform.

**Comment by gedankenstuecke** [comment]
Sure, then I'd say we go with `tinyverse`, which has all the packages you want.

**Comment by madprime** [comment]
For Python I wondered if something enabling "import from a remote URL" might exist –
 and found this. I think it'd be a great addition, assuming it works as it promises?
 https://github.com/operatorequals/httpimport

(And a comment – although the README refers to importing from a github "master" branch URL, for security reasons, I'd generally encourage imports from URLs that refer to a specific commit hash.)

**Comment by betatim** [comment]
Add `pip install open-humans-api` to the image.
